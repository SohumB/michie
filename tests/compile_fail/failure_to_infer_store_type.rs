use michie::memoized;

#[memoized(key_expr = (), store_init = Default::default())]
fn f() -> () {}

fn main() {}
