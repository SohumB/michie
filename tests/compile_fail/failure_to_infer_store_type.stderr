error[E0282]: type annotations needed for `&_`
 --> tests/compile_fail/failure_to_infer_store_type.rs:3:1
  |
3 | #[memoized(key_expr = (), store_init = Default::default())]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ consider giving `cache` the explicit type `&_`, with the type parameters specified
  |
  = note: this error originates in the attribute macro `memoized` (in Nightly builds, run with -Z macro-backtrace for more info)
